src/main/java/com/amazon/aocagent/services/ECSService.java

```diff
@@ -0,0 +1,354 @@
+  private static final String ECS_PREFIX = "ecs";
+  private static final String AOC_EMITTER = "aoc-emitter";
+  private static final String EC2 = "EC2";
+  private static final String INSTANCE_ID = "otlp_instance_id";
+  private static final String OTLP_ENDPOINT = "otlp_endpoint";
+  private AmazonECS ecsClient;
+  private AWSLogs awsLogsClient;
+
+  public ECSService(final String region) {
+    this.ecsClient = AmazonECSClientBuilder.standard().withRegion(region).build();
+    this.awsLogsClient = AWSLogsAsyncClientBuilder.standard().withRegion(region).build();
+  }
+
+  /**
+   * create ECS cluster for integ testing.
+   * @return cluster creation result
+   */
+  public CreateClusterResult createCluster() {
+    CreateClusterRequest request =
+        new CreateClusterRequest().withClusterName(GenericConstants.ECS_SIDECAR_CLUSTER.getVal());
+    log.info("creating ECS cluster: {}", GenericConstants.ECS_SIDECAR_CLUSTER.getVal());
+    return ecsClient.createCluster(request);
+  }
+
+  /**
+   * create and register ecs task definition for both EC2 and fargate running mode.
+   * @param context test context
+   * @return {@link RegisterTaskDefinitionResult} for registering ecs container defs
+   */
+  public RegisterTaskDefinitionResult registerTaskDefinition(Context context) {
+    com.amazonaws.services.ecs.model.RegisterTaskDefinitionRequest request;
+
+    String launchType = context.getLaunchType();
+    List<ContainerDefinition> containerDefinitionList = this.buildContainerDefinition(context);
+    if (launchType.equalsIgnoreCase(EC2)) {
+      request =
+          new RegisterTaskDefinitionRequest()
+              .withFamily("aoc-" + launchType)
+              .withTaskRoleArn(GenericConstants.ECS_TASK_ROLE.getVal())
+              .withExecutionRoleArn(GenericConstants.ECS_TASK_EXECUTION_ROLE.getVal())
+              .withRequiresCompatibilities(Compatibility.EC2)
+              .withNetworkMode(NetworkMode.Bridge)
+              .withCpu("256")
+              .withMemory("512")
+              .withContainerDefinitions(containerDefinitionList);
+
+    } else {
+      request =
+          new RegisterTaskDefinitionRequest()
+              .withFamily("aoc-" + launchType)
+              .withTaskRoleArn(GenericConstants.ECS_TASK_ROLE.getVal())
+              .withExecutionRoleArn(GenericConstants.ECS_TASK_EXECUTION_ROLE.getVal())
+              .withRequiresCompatibilities(Compatibility.FARGATE)
+              .withNetworkMode(NetworkMode.Awsvpc)
+              .withCpu("256")
+              .withMemory("512")
+              .withContainerDefinitions(containerDefinitionList);
+    }
+    return ecsClient.registerTaskDefinition(request);
+  }
+
+  private List<ContainerDefinition> buildContainerDefinition(Context context) {
+    List<ContainerDefinition> definitions = new ArrayList<>();
+
+    PortMapping portMapping =
+        new PortMapping()
+            .withContainerPort(55680)
```

Comment:
why port mapping is needed if just for sidecar test?

Reply:
this port is for OTLP receiver in AOC. I think it is for all the cases.

Reply:
can we move OT logic outside from service level? Decoupling these logics could help us to expand more testing in the future.